### ISSUE TITLE
Quantized network will load failed  if contains argminmax layer

### ISSUE THREAD

**Issue by ek9852** [ARM-software/armnn#490]
We want to keep QAsymmU8 as input, but this makes output have to be QAsymmU8 also in ArmnnQuantizer. When the network is loaded by arm deserializer, it just  silently catch this exception without any log to the console:
        throw InvalidArgumentException(descriptorName + ": Output of ArgMinMax layer must be Int32 or Int64.");
        
  We should have argminmax to support QAsymmU,  QAsymmS8 and QSymm16.
  Or have a choice to have output format as int32 but input format as QAsymmU8 in ArmnnQuantizer . Right now both input and output format must be same in ArmnnQuantizer.

**Comment by MikeJKelly** [comment]
Hi @ek9852 

where are you getting your model from? Can you share it? The output of an argmin/argmax layer has to be either int32 or int64 because argmin/argmax return the index of the smallest or largest element along an axis, not the value of the element.

Best regards,
Mike

**Comment by ek9852** [comment]
I can't share the model, but it is just a unet model with argmax at the end to do semantic segmentation.  When I am running ArmnnQuantizer , i did not use preserve input and output type, so  ArmnnQuantizer force the output from argmax to be in QAsymmU8 also. ArmnnQuantizer does not output any error during the conversion, but there is an exception when the serialized armnn model file  is load with "Output of ArgMinMax layer must be Int32 or Int64"

**Comment by MikeJKelly** [comment]
Hi @ek9852 

I just wanted to share some information with you. We have no plans to develop either the Caffe parser or the ArmnnQuantizer further. It's possible that both the Caffe parser and ArmnnQuantizer may be dropped from Armnn in the near future. 

Caffe is an older framework that is no longer being developed and we’ve seen a shift to other frameworks.

Our quantizer implementation was intended as a stopgap until other quantization toolkits became available, you might want to look at the quantization tools in TensorFlow or PyTorch.  

Best regards,
Mike

**Comment by MikeJKelly** [comment]
Another note on this: Although we're not going to be adding new features I am looking at the bug you've reported here where the ArmnnQuantizer incorrectly changes the output of argmax/argmin to be QAsymmU8 and hope to have a fix for it soon.

**Comment by ek9852** [comment]
U will encounter same problem for onnx. It is not a caffe specific bug.
Once u added back the missing argminmax layer support to onnx.
Since the argmax layer in compute library only output int32, not int8 or uint7.

BTW deconv support still missing in onnx parser, but I think i should fire a new issue for that.

**Comment by ek9852** [comment]
Fixed by https://github.com/ARM-software/armnn/commit/c5e5a04bd26c2d1838828ba291464967a3e20f3f
